src/main/java/com/ericsson/ei/notifications/EmailSender.java

```diff
@@ -90,6 +90,19 @@ public MimeMessage prepareEmailMessage(String recipients, String mapNotification
         return message;
     }
 
+    /*
+     * This method assigns the default values for sender and subject of email notification if values
+     * at application.proporties are empty.
+     * */
+    private void setDefaultValuesEmailSenderSubject() {
+       if (subject.isEmpty()) {
+            subject = "Email Subscription Notification";
```

Comment:
Is this if-statements really needed since it set default to  "Email Subscription Notification" on line 55, if subject string is empty.
So whole this if-statement can be removed, if understand the code correctly.

Reply:
Apparently Spring doesn't fill in default values if the property is present in application.properties but without values. It interprets "" as a value and will not override it ... So that's why this check is needed. Perhaps add this in the javadoc for the method to clarify future questions 😄

Reply:
yah as per my testing I observed that even the values are given at @value in Emailsender class, those are not been assigned  when the user gives as below in application.proporties.
email.sender:
email.subject:

when I tried to print sender and subject.. still I can see those empty values.
so I have kept those if statments to handle isEmpty values for sender and subject.

This @Value("${email.sender:noreply@domain.com}") and @Value("${email.subject:Email Subscription Notification}") line is working only when the entire field email.sender and email.subject are not present in application.proporties.

Reply:
I saw the javadoc was updated in a previous commit, nice!

Reply:
Then I guess default value in @Value ":noreply@....." is not needed , since its defaulted here instead.

Reply:
Right now its not needed , I just kept it considering the situation (if in case the fields email.sender and email.subject) are not present in application.proporties